### ISSUE TITLE
Place 'edit' and ‘remove’ functionalities within the same menu tier

### ISSUE THREAD

**Issue by cooperzoe** [OpenRefine/OpenRefine#6280]
Currently, you have to navigate to 'edit column' first in order to remove a column. I found it a bit counterintuitive to have ‘remove’ functionality as a subset of the 'edit' functionality; to me, they are two separate kinds of actions on par with one another.

In other words: I’m either looking to edit a column or to delete it. I wouldn't think to edit a column I'm intending to get rid of, so navigating first to 'edit' in this instance feels a bit unclear. I think they should appear on the same level. 

Proposed solution: I'd like to propose adding the remove column functionality - or any kind of functionality that involves deleting information - to the first menu that pops up, not the second.

Here's a screenshot for reference:
![Screenshot 2024-01-04 at 10 54 13 AM](https://github.com/OpenRefine/OpenRefine/assets/52141007/3f223c2d-cce0-4805-8aae-c8ad01c66684)

**Comment by thadguidry** [comment]
Agree.  Remove and Rename are used quite often from my trainings I've seen in the past.  Remove could even maybe have a dedicated small X button on each column header?  Rename could be performed by double-click on the column header that makes it editable, but I feel that style of renaming a column sometimes gets in the way and accidental click/renames happen, based on my experience in other apps that do that.

**Comment by smierz** [comment]
hey, i noticed the "good first issue" tag here, is this a simple change of moving the "Remove this column" line from the edit-submenu to the edit-menu (between "Edit column" and "Transpose")?
If that's the case, I could work on it.

**Comment by thadguidry** [comment]
Hi @smierz Yes, but...
I personally think both Rename and Remove should be moved there.
But let's first get consensus from @wetneb @ostephens @magdmartin

**Comment by wetneb** [comment]
Also moving "Rename" makes sense to me, they intuitively belong together I think.

**Comment by smierz** [comment]
Here's a preview of simply moving the entries one menu up and removing the icons to make the look more coherent:

Before             |  After
:-------------------------:|:-------------------------:
![Image](https://github.com/user-attachments/assets/2ec47d25-941b-42d0-aa90-82982b30c0bd)  | ![Image](https://github.com/user-attachments/assets/c687d04e-e3ee-411d-9604-7572ee169f88)

Note: it would probably make sense to additionally shorten the menu entries by removing "this", e.g. "Rename column.." instead of "Rename **this** column" which would make them fit into one line
--> but this would need adjustment through all translation files

**Comment by thadguidry** [comment]
I like it!  And agree to shorten in the English `en` translation file.

**Comment by wetneb** [comment]
You don't need to worry about translations: your PR can just change the English ones, and translators will be notified of the change on their end.

**Comment by smierz** [comment]
I opened a PR with the changes already discussed in this thread.

**Open question: icons yes or no**
As stated above I removed the icons in front of the rename/remove column operations to make the look of the menu more coherent (see screenshot in linked PR).

I also took screenshots for the options to include the icons:
1. include icons without resize -->  splits the text into two lines
![Image](https://github.com/user-attachments/assets/ece60f15-60c7-40bc-824c-a844f2861444)
2. include icons with resize of the menu-container (+15px) --> text and icons fit into one line
![Image](https://github.com/user-attachments/assets/1f1d2bfc-1ea6-4109-9454-7f08bdb794da)

Let me know what you think/prefer!

**Comment by thadguidry** [comment]
I like them on one line.  That's also how Excel, Google Sheets, LibreOffice and others look and feel.

**Comment by tfmorris** [comment]
Having lived with this for a little while, I've failed to learn to love having the column operations split into two separate locations. I  think a better solution to the OP's cognizant dissonance might be to rename "Edit column" to "Column operations" (or just "Column...").

Having dedicated affordances in the header for delete & rename would improve access for those functions, but "Rename column," to my mind, actually is a form of "Edit Column." Conversely, the various "Add column" operations aren't, so the logical argument doesn't really hold.

I would suggest:
- moving the delete & rename column items back where they were
- renaming "Edit column" to something that doesn't cause cognitive dissonance
- considering adding dedicated affordances/actions for column rename and delete

**Comment by tfmorris** [comment]
Reopening this so that we can consider whether we really want to include it in the final 3.10 release.

Concrete proposal for an alternative solution to the original complaint:
- Change "Edit column ->" to "Column operations ->" (or just "Column... ->")
- Return Rename and Delete to the submenu with the rest of the column operations

**Comment by magdmartin** [comment]
There’s an active forum poll on testing OpenRefine 3.10 and feedback on the new column menu. 
Pease share your input here: [https://forum.openrefine.org/t/poll-3-10-testing-new-column-menu/2703](https://forum.openrefine.org/t/poll-3-10-testing-new-column-menu/2703)

**Comment by magdmartin** [comment]
So, after testing the new release, I do prefer the 3.10 column menu.
* It better supports discoverability, which is one of the main motivations discussed above
  * “[Easier rename a column by clicking on it”](https://openrefine.org/docs/technical-reference/goal-posts#easier-rename-a-column-by-clicking-on-it) appears among the top-ranked feature ideas in the 2024 survey, suggesting that the 3.9 structure introduces friction around common column-level actions.
  * This change also aligns with our [values](https://openrefine.org/mission_vision) of being `Approachable` and `Transparent & Open`, by making common actions more visible and improving onboarding for new users (with a one-time muscle-memory adjustment for long-time users).
* It improves speed of execution for high-frequency actions like renaming and removing columns when preparing or reorganising a dataset.
